### ISSUE TITLE
Unable to add new lanes - Any edit in the fast file are not recognised.

### ISSUE THREAD

**Issue by akshaynhegde** [fastlane/fastlane#6168]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

Any change I make in the fast file are not recognised. The new lane am trying to adding is not getting recognised. I added an empty lane called "tryout", and am unable to run that lane because its not found. Even if I change the description of predefined lanes (test, beta, release) it does not get reflected when I run `fast lane lanes` 

Would be great if you could help me figure out whats going wrong.

##### Complete output when running fastlane, including the stack trace and command used

```
akshays-MacBook-Pro:SignEasy akshay$ fastlane tryout
[18:44:57]: -------------------------------------------------
[18:44:57]: --- Step: Verifying required fastlane version ---
[18:44:57]: -------------------------------------------------
[18:44:57]: fastlane version valid
[18:44:57]: ------------------------------
[18:44:57]: --- Step: default_platform ---
[18:44:57]: ------------------------------
[18:44:58]: Variable Dump:
[18:44:58]: {:DEFAULT_PLATFORM=>:ios}
[18:44:58]: Could not find lane 'ios tryout'. Available lanes: ios test, ios beta, ios release

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
+------+-------------------------------------+-------------+

[18:44:58]: fastlane finished with errors

[!] Could not find lane 'ios tryout'. Available lanes: ios test, ios beta, ios release
```

### Configuration Files

Please copy the complete content of your `Fastfile` and any other configuration files you use below:

Fastfile:
```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "1.103.0"

default_platform :ios

platform :ios do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
    cocoapods
    
  end

  desc "Runs all the tests"
  lane :test do
    scan
  end
  
  desc "Try out Lane"
  lane :tryout do
      # empty lane, testing if the new lane gets detected.
  end

  desc "Submit a new Beta Build to Apple TestFlight"
  desc "This will also make sure the profile is up to date"
  lane :beta do
    # match(type: "appstore") # more information: https://codesigning.guide
    gym(scheme: "SignEasy") # Build your app - more options available
    pilot

    # sh "your_script.sh"
    # You can also use other beta testing services here (run `fastlane actions`)
  end

  desc "Deploy a new version to the App Store"
  lane :release do
    # match(type: "appstore")
    # snapshot
    gym(scheme: "SignEasy") # Build your app - more options available
    deliver(force: true)
    # frameit
  end

  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end

  error do |lane, exception|
    # slack(
    #   message: exception.message,
    #   success: false
    # )
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer
```

### Environment

fastlane version (run `fastlane -v`):

fastlane 1.103.0

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)?

No

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)?

No

**Comment by KrauseFx** [comment]
I just gave your Fastfile a try it's working just expected. Are you sure you're editing the right file?

**Comment by akshaynhegde** [comment]
Its the same fastfile that got created in after fastlane init. I just included it in the xcode project and editing it.  So yes, it must be the same file.

**Comment by akshaynhegde** [comment]
Hey Felix,
Just deleted the entire fastlane folder. Re run `fastlane init` . Here is the output (only last few lines)

```
[10:30:54]: Downloading existing screenshot '3_ipadPro_3.Finnish_3.jpg'
[10:30:56]: Downloading existing screenshot '4_ipadPro_4.Finnish_4.jpg'
[10:30:57]: Downloading existing screenshot '5_ipadPro_5.Finnish_5.jpg'
[10:30:59]: Successfully downloaded all existing screenshots
[10:30:59]: Successfully created new Deliverfile at path 'fastlane/Deliverfile'
[10:30:59]: 'snapshot' not enabled.
[10:30:59]: 'cocoapods' enabled.
[10:30:59]: 'carthage' not enabled.
[10:30:59]: Created new file './fastlane/Fastfile'. Edit it to manage your own deployment lanes.
[10:30:59]: fastlane will send the number of errors for each action to
[10:30:59]: https://github.com/fastlane/enhancer to detect integration issues
[10:30:59]: No sensitive/private information will be uploaded
[10:30:59]: Successfully finished setting up fastlane
```

Then I see a fastlane folder in my repo. Included it in teh xcode project. Just modified the description of one of the lanes. And when I run `fastlane lanes` , I see only the default generated lanes and none of my modifications

```
akshays-MacBook-Pro:SignEasy akshay$ fastlane lanes
[10:41:39]: -------------------------------------------------
[10:41:39]: --- Step: Verifying required fastlane version ---
[10:41:39]: -------------------------------------------------
[10:41:39]: fastlane version valid
[10:41:39]: ------------------------------
[10:41:39]: --- Step: default_platform ---
[10:41:39]: ------------------------------

--------- ios---------
----- fastlane ios test
Runs all the tests

----- fastlane ios beta
Submit a new Beta Build to Apple TestFlight
This will also make sure the profile is up to date

----- fastlane ios release
Deploy a new version to the App Store

Execute using `fastlane [lane_name]`

```

So, am I missing something here?

**Comment by TKBurner** [comment]
Hi @akshaynhegde! Thanks for following up. What lane are you expecting to see here? Your Fastfile only has three lanes. Thanks!

**Comment by KrauseFx** [comment]
There is no `tryout` lane in your Fastfile you posted a screenshot of

**Comment by akshaynhegde** [comment]
@TKBurner @KrauseFx , Sorry for the wrong screenshot.  Attaching the right one now. 

I have made two changes to the file.
1. Updated the description of the "test" lane
2. Added an empty "tryout" lane (Totally empty)

When I run `fastlane lanes`, I see the following. Please note, that I am inside my project folder. And the command did not print out neither the updated description nor the new lane.

I have a feeling that I might be missing something really stupid. But am not able to figure out what. The init was successful. I see the fastfile under my repo. Same file is added to the project which am trying to edit.

```
akshays-MacBook-Pro:~ akshay$ cd /Users/akshay/GIT/SignEasy/iOS-App-v2/SignEasy 
akshays-MacBook-Pro:SignEasy akshay$ fastlane lanes
[07:23:21]: -------------------------------------------------
[07:23:21]: --- Step: Verifying required fastlane version ---
[07:23:21]: -------------------------------------------------
[07:23:21]: fastlane version valid
[07:23:21]: ------------------------------
[07:23:21]: --- Step: default_platform ---
[07:23:21]: ------------------------------

--------- ios---------
----- fastlane ios test
Runs all the tests

----- fastlane ios beta
Submit a new Beta Build to Apple TestFlight
This will also make sure the profile is up to date

----- fastlane ios release
Deploy a new version to the App Store

Execute using `fastlane [lane_name]`


```
<img width="939" alt="screen shot 2016-09-23 at 7 23 36 am" src="https://cloud.githubusercontent.com/assets/3814615/18771727/cc0c5b04-815e-11e6-9e03-bcb6b344dbfd.png">

**Comment by akshaynhegde** [comment]
@TKBurner @KrauseFx ,

I wanted to double check and eliminate the stupid mistakes and  It indeed was a stupid one!. 
I tried to make sure if the fastfile in the project and in the repo are same,

I was adding the fastfile inside the main group/folder (SignEasy). It was creating a copy of the fastfile when added to the project. All I had to do was to add the fastfile outside it in its right place and everything is working as expected.

Thank you so much for all the help :)

<img width="940" alt="screen shot 2016-09-23 at 7 42 39 am" src="https://cloud.githubusercontent.com/assets/3814615/18772100/bb1c6c96-8161-11e6-8d12-cb18b702258e.png">
